changelogs/unreleased/gh-9924-luajit-fixes.md

```diff
@@ -0,0 +1,17 @@
+## bugfix/luajit
+
+Backported patches from the vanilla LuaJIT trunk (gh-9924). The following issues
+were fixed as part of this activity:
+
```

Comment:
I would also mention added UBSan support and seems commit "Prevent sanitizer warning in snap_restoredata()." is missing in changelog.

Reply:
Since UBSan support is not a user-visible change (it is more for our testing), it is omitted in the changelog. The sanitizer warning in `snap_restoredata()` is also harmless since it can't lead to any misbehaviour, so it is not visible to the user either.